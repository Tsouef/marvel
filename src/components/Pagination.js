import React, { Component } from 'react';
import { Link } from 'react-router-dom';

class Pagination extends Component {
  state = {
    letters: [
      'a',
      'b',
      'c',
      'd',
      'e',
      'f',
      'g',
      'h',
      'i',
      'j',
      'k',
      'l',
      'm',
      'n',
      'o',
      'p',
      'q',
      'r',
      's',
      't',
      'u',
      'v',
      'w',
      'x',
      'y',
      'z'
    ]
  };

  renderContent() {
    return this.state.letters.map(letter => {
      return (
        <li key={letter}>
          <Link to={`/characters/letter/${letter}`}>{letter}</Link>
        </li>
      );
    });
  }

  render() {
    return <ul className="pagination center-align">{this.renderContent()}</ul>;
  }
}

export default Pagination;
