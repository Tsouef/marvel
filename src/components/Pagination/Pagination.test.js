import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';

import React from 'react';
import Pagination from './Pagination';

describe('component <Pagination />', () => {
  it('should render Pagination component', () => {
    const props = {
      category: 'comics'
    };
    const tree = renderer
      .create(
        <MemoryRouter>
          <Pagination {...props} />
        </MemoryRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
