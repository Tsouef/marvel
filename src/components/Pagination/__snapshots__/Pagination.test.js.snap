// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component <Pagination /> should render Pagination component 1`] = `
<ul
  className="pagination center-align"
>
  <li>
    <a
      href="/comics/letter/a"
      onClick={[Function]}
    >
      a
    </a>
  </li>
  <li>
    <a
      href="/comics/letter/b"
      onClick={[Function]}
    >
      b
    </a>
  </li>
  <li>
    <a
      href="/comics/letter/c"
      onClick={[Function]}
    >
      c
    </a>
  </li>
  <li>
    <a
      href="/comics/letter/d"
      onClick={[Function]}
    >
      d
    </a>
  </li>
  <li>
    <a
      href="/comics/letter/e"
      onClick={[Function]}
    >
      e
    </a>
  </li>
  <li>
    <a
      href="/comics/letter/f"
      onClick={[Function]}
    >
      f
    </a>
  </li>
  <li>
    <a
      href="/comics/letter/g"
      onClick={[Function]}
    >
      g
    </a>
  </li>
  <li>
    <a
      href="/comics/letter/h"
      onClick={[Function]}
    >
      h
    </a>
  </li>
  <li>
    <a
      href="/comics/letter/i"
      onClick={[Function]}
    >
      i
    </a>
  </li>
  <li>
    <a
      href="/comics/letter/j"
      onClick={[Function]}
    >
      j
    </a>
  </li>
  <li>
    <a
      href="/comics/letter/k"
      onClick={[Function]}
    >
      k
    </a>
  </li>
  <li>
    <a
      href="/comics/letter/l"
      onClick={[Function]}
    >
      l
    </a>
  </li>
  <li>
    <a
      href="/comics/letter/m"
      onClick={[Function]}
    >
      m
    </a>
  </li>
  <li>
    <a
      href="/comics/letter/n"
      onClick={[Function]}
    >
      n
    </a>
  </li>
  <li>
    <a
      href="/comics/letter/o"
      onClick={[Function]}
    >
      o
    </a>
  </li>
  <li>
    <a
      href="/comics/letter/p"
      onClick={[Function]}
    >
      p
    </a>
  </li>
  <li>
    <a
      href="/comics/letter/q"
      onClick={[Function]}
    >
      q
    </a>
  </li>
  <li>
    <a
      href="/comics/letter/r"
      onClick={[Function]}
    >
      r
    </a>
  </li>
  <li>
    <a
      href="/comics/letter/s"
      onClick={[Function]}
    >
      s
    </a>
  </li>
  <li>
    <a
      href="/comics/letter/t"
      onClick={[Function]}
    >
      t
    </a>
  </li>
  <li>
    <a
      href="/comics/letter/u"
      onClick={[Function]}
    >
      u
    </a>
  </li>
  <li>
    <a
      href="/comics/letter/v"
      onClick={[Function]}
    >
      v
    </a>
  </li>
  <li>
    <a
      href="/comics/letter/w"
      onClick={[Function]}
    >
      w
    </a>
  </li>
  <li>
    <a
      href="/comics/letter/x"
      onClick={[Function]}
    >
      x
    </a>
  </li>
  <li>
    <a
      href="/comics/letter/y"
      onClick={[Function]}
    >
      y
    </a>
  </li>
  <li>
    <a
      href="/comics/letter/z"
      onClick={[Function]}
    >
      z
    </a>
  </li>
</ul>
`;
